`sccache` marked as Linux-only

**Describe the bug**

It looks like the `sccache-dist` was added to the `sccache` package, which is Linux-only. Later, a PR was accepted that marked the entirety of `sccache` as Linux-only when it absolutely works on macOS and probably (though I don't run them) BSD variants. This seems like an incorrect way to include the `sccache-dist` utility and instead should have been included as a separate package to avoid breaking this package for non-Linux operating systems.

https://github.com/NixOS/nixpkgs/commit/2a0d3cc90764141a076995534a04be9f17e0c40d
https://github.com/NixOS/nixpkgs/commit/a094d0e0e4804dfa2a28eb81b4db683aee2d2604

**To Reproduce**
Steps to reproduce the behavior:
1. Install `sccache` on a macOS machine with an up-to-date unstable channel.

**Expected behavior**
`sccache` should be installable on its supported platforms, in this case `macOS`.

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.3.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.10`
 - channels(david): `"darwin, home-manager, nixpkgs-21.05pre274251.f5f6dc053b1"`
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
 - nixpkgs: `/Users/david/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - sccache
# a list of nixos modules affected by the problem
module:
```

Build currently fails on darwin:
```
error[E0432]: unresolved import `nix::mount`
 --> /private/tmp/nix-build-sccache-0.2.15.drv-0/sccache-0.2.15-vendor.tar.gz/libmount/src/bind.rs:6:10
  |
6 | use nix::mount::{MsFlags, mount};
  |          ^^^^^ could not find `mount` in `nix`

error[E0432]: unresolved import `nix::mount`
 --> /private/tmp/nix-build-sccache-0.2.15.drv-0/sccache-0.2.15-vendor.tar.gz/libmount/src/overlay.rs:8:10
  |
8 | use nix::mount::{MsFlags, mount};
  |          ^^^^^ could not find `mount` in `nix`

error[E0432]: unresolved import `nix::mount`
 --> /private/tmp/nix-build-sccache-0.2.15.drv-0/sccache-0.2.15-vendor.tar.gz/libmount/src/tmpfs.rs:8:10
  |
8 | use nix::mount::{MsFlags, mount};
  |          ^^^^^ could not find `mount` in `nix`

error[E0432]: unresolved import `nix::mount`
 --> /private/tmp/nix-build-sccache-0.2.15.drv-0/sccache-0.2.15-vendor.tar.gz/libmount/src/modify.rs:5:10
  |
5 | use nix::mount::{MsFlags, mount};
  |          ^^^^^ could not find `mount` in `nix`

error[E0432]: unresolved import `nix::mount`
  --> /private/tmp/nix-build-sccache-0.2.15.drv-0/sccache-0.2.15-vendor.tar.gz/libmount/src/remount.rs:10:10
   |
10 | use nix::mount::{MsFlags, mount};
   |          ^^^^^ could not find `mount` in `nix`

error[E0432]: unresolved import `nix::mount`
  --> /private/tmp/nix-build-sccache-0.2.15.drv-0/sccache-0.2.15-vendor.tar.gz/libmount/src/mountinfo.rs:10:10
   |
10 | use nix::mount::MsFlags;
   |          ^^^^^ could not find `mount` in `nix`

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0432`.
error: could not compile `libmount`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/pkgs/development/tools/misc/sccache/default.nix
+++ b/pkgs/development/tools/misc/sccache/default.nix
@@ -13,7 +13,7 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "1cfdwf00jgwsv0f72427asid1xr57s56jk5xj489dgppvgy7wdbj";
 
-  cargoBuildFlags = [ "--features=dist-client,dist-server" ];
+  cargoBuildFlags = lib.optionals (!stdenv.isDarwin) [ "--features=dist-client,dist-server" ];
 
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [ openssl ] ++ lib.optional stdenv.isDarwin Security;
@@ -27,6 +27,5 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/mozilla/sccache";
     maintainers = with maintainers; [ doronbehar ];
     license = licenses.asl20;
-    platforms = [ "x86_64-linux" ];
   };
 }

